File: 0227-basic-calculator-ii/0227-basic-calculator-ii.py
class Solution:
    def calculate(self, s: str) -> int:
        number = 0
        operator = "+"
        i=0
        stack = []
        while i<len(s):
            if(s[i].isdigit()):
                number = number * 10 + int(s[i])

            if (s[i] == "+" or s[i] == "-" or s[i] == "*" or s[i] == "/") or i == len(s)-1:
                #operator
                if operator == "+":
                    stack.append(number)
                elif operator == "-":
                    stack.append(-number)
                elif operator == "*":
                    stack.append(stack.pop() * number)
                elif operator == "/":
                    stack.append(int(stack.pop() / number))

                number = 0
                operator = s[i]
            i += 1
        
        res = 0
        while len(stack) > 0:
            res += stack.pop()
        
        return res

File: 0227-basic-calculator-ii/0227-basic-calculator-ii.js
/**
 * @param {string} s
 * @return {number}
 */
var calculate = function(s) {
    let number = 0
    let op = "+"
    let i=0
    const stack = []
    while(i<s.length) {
        const ch = s[i]
        if(ch.charCodeAt() >= 48 && ch.charCodeAt() <= 57) {
            number = number * 10 + Number(ch)
        }
        console.log("number: ", number)
        if(ch === "+" || ch === "-" || ch === "*" || ch === "/" || i === s.length-1) {
            if(op === '+') {
                stack.push(number)
            } else if(op === '-') {
                stack.push(-number)
            } else if(op === "*") {
                stack.push(stack.pop() * number)
            } else if(op === "/") {
                stack.push(parseInt(stack.pop() / number))
            }
            op = ch
            number = 0
        }
        i++
    }
    let res = 0;
    while (stack.length > 0) {
        res += stack.pop()
    }
    return res
};